etl/src/v2/state/store/postgres.rs

```diff
@@ -0,0 +1,269 @@
+use std::{collections::HashMap, sync::Arc};
+
+use config::shared::{SourceConfig, TlsConfig};
+use postgres::schema::TableId;
+use sqlx::{
+    postgres::{types::Oid as SqlxOid, PgConnectOptions, PgPoolOptions, PgSslMode},
+    prelude::{FromRow, Type},
+    PgPool, Row,
+};
+use thiserror::Error;
+use tokio::sync::RwLock;
+use tokio_postgres::types::PgLsn;
+
+use crate::v2::{
+    pipeline::PipelineId,
+    replication::slot::get_slot_name,
+    state::{
+        store::base::{StateStore, StateStoreError},
+        table::TableReplicationPhase,
+    },
+    workers::base::WorkerType,
+};
+
+#[derive(Debug, Clone, Copy, Type, PartialEq)]
+#[sqlx(type_name = "etl.table_state", rename_all = "snake_case")]
+pub enum TableState {
+    Init,
+    DataSync,
+    FinishedCopy,
+    SyncDone,
+    Ready,
+    Skipped,
+}
+
+#[derive(Debug, Error)]
+pub enum ToTableStateError {
+    #[error("In-memory table replication phase can't be saved in the state store")]
+    InMemoryPhase,
+}
+
+impl TryFrom<TableReplicationPhase> for TableState {
+    type Error = ToTableStateError;
+
+    fn try_from(value: TableReplicationPhase) -> Result<Self, Self::Error> {
+        Ok(match value {
+            TableReplicationPhase::Init => TableState::Init,
+            TableReplicationPhase::DataSync => TableState::DataSync,
+            TableReplicationPhase::FinishedCopy => TableState::FinishedCopy,
+            TableReplicationPhase::SyncDone { .. } => TableState::SyncDone,
+            TableReplicationPhase::Ready => TableState::Ready,
+            TableReplicationPhase::Skipped => TableState::Skipped,
+            _ => return Err(ToTableStateError::InMemoryPhase),
```

Comment:
A general pattern in Rust is to manually specify the types and avoid using `_` (when possible), since if a new phase is added, we want the user to explicitly handle its conversion. Otherwise, with the last branch, this might go unnoticed and cause issues.